testing/asserts_test.ts

```diff
@@ -487,6 +490,20 @@ Deno.test("testingAssertObjectMatching", function (): void {
     }
     assertEquals(didThrow, true);
   }
+  // Subset with array inside but doesn't match key subset
+  {
+    let didThrow;
+    try {
+      assertObjectMatch(h, {
+        foo: [],
+      });
```

Comment:
I feel this might better to pass the assertion because everything in `{ foo: [] }` is contained in h (`{ foo: [1, 2, 3], bar: true }`). Maybe we should do some recursive check for array as well?

Reply:
Yes it makes sense, will make the change 👍

Reply:
I changed it so `{ foo: [1, 2, 3], bar: true }` matches `{ foo: [] }` as suggested
When specifying complex arrays like `{ foo: [ { bar: true, baz: false } ] }` it'll match both `{ foo: [ { bar:true } ] }` and `{ foo: [ { baz: false } ] }`